routes/index.js

```diff
@@ -116,13 +122,47 @@ const handlePublicChannel = (req, res) => {
   const chatName = req.params[0]
   genPage(req, res, {
     title: `Join #${chatName} in Status`,
-    info: `Join public channel <span class="inline-block align-bottom w-32 truncate">#${chatName}</span> on Status.`,
+    info: `Join public channel <span class="inline-block align-bottom w-32 truncate">#${chatName}</span> in Status.`,
     mainTarget: chatName, 
     headerName: `#${chatName}`,
     path: req.originalUrl,
   })
 }
 
+/* This verifies all 3 required URL arguments are present */
+const validatePrivateGroupArgs = (args) => {
+  const requiredKeys = ['a', 'a1', 'a2']
+  const argsHasKey = key => Object.keys(args).includes(key)
+  if (!requiredKeys.every(argsHasKey)) {
+    throw Error('Missing arguments!')
+  }
+  if (args.a == null || args.a.length != 132) {
+    throw Error('Admin public key invalid!')
+  }
+  if (args.a2 == null || args.a2.length != 169) {
+    throw Error('Chat public key invalid!')
+  }
+}
+
+/* Open Private Group in Status */
+const handlePrivateGroup = (req, res) => {
+  try {
+    validatePrivateGroupArgs(req.query)
+  } catch(ex) {
+    handleError(`Invalid private group URL: ${ex.message}`)(req, res)
+    return
+  }
+  let groupName = req.query.a1
+  genPage(req, res, {
+    title: `Join "${groupName}" private group in Status`,
+    info: `Join private group <span class="inline-block align-bottom w-32 truncate">${groupName}</span> in Status.`,
```

Comment:
in status we call it `Group chat` so probably it should be `Join group chat` ?

Reply:
Oh, I see. Sure, let me change that.

Reply:
Fixed.